content/blog/options/index.md

```diff
@@ -0,0 +1,237 @@
+
+This just defers the problem. 
+The user of the function will get a panic at runtime. It might be your future self.
+
+Moreover, I advise people to avoid `unwrap` in production code, but this sets a bad example.
+
+During trainings, this is where I pause for a moment and explain the error message.
+It's a great point where we dig deeper to understand the root cause of the problem.
+
+Okay, I've kept you waiting long enough. Let's demystify this error message.
+
+## Why doesn't `?` work with `Option`?
+
+The `?` operator takes a `Result` and unwraps it, returning the value inside if it is `Ok`, or returning early if it is `Err`. 
+
+Essentially, `?` is implemented like this:
+
+```rust
+let value = match my_result {
+    Ok(value) => value,
+    Err(error) => return Err(error),
+};
+```
+
+`Option` on the other hand **does not have an** `Err` variant. 
+
+It is defined like this:
+
+```rust
+enum Option<T> {
+    Some(T),
+    None,
+}
+```
+
+`None` doesn't have an associated error value. It's just a value that means "no value".
+The `?` operator wouldn't know what error it should return:
+
+```rust
+let value = match my_option {
+    Some(value) => value,
+    None => todo!("What to return here? ¯\_(ツ)_/¯"),
+};
+```
+
+That's actually the entire difference between `Result` and `Option`.
+If we had an `Err` variant, we would just use `Result` in the first place.
+But there are cases where `None` is a completely valid value and we just don't have an error to return.
+We can just say "there is no value" and that's it.
+
+## What's The Solution?
+
+There are multiple solutions!
+
+The initial error message, while cryptic, gave us a hint:
+
+```rust
+use `.ok_or(...)?` to provide an error compatible with `Result<(), Box<dyn std::error::Error>>`.
+```
+
+Apparently we can use the `ok_or` method, which converts the `Option` into a `Result`:
+
+```rust
+let user = get_user().ok_or("No user")?;
+```
+
+Unfortunately, `ok_or` might not be the most discoverable name for this method.
+I find it unintuitive and needed to look it up many times.
+That's because `Ok` is commonly associated with the `Result` type, not `Option`.
+There's `Option::Some`, so it could also be called `some_or`, but that would be hard to change now.
+It was [actually suggested in 2014](https://github.com/rust-lang/rust/pull/17469#issuecomment-56919911), but the name `ok_or` won out,
+because `ok_or(MyError)` reads nicely and I can see that. Guess we have to live with the minor inconsistency now.
+
+Another downside is that `ok_or` *always* allocates a new `String` for the error message &ndash; even if we don't use it.
+If you want to avoid this allocation, you can use `ok_or_else`:
+
+```rust
+let user = get_user().ok_or_else(|| "No user".into())?;
+```
+
+This takes a closure that returns the error message.
+The closure only gets called if `get_user()` returns `None`.
+
+We avoid the allocation if not needed, but that's even more verbose!
```

Comment:
Yes, true. And a nice little detail. But it also distracts from the main point of this post. I'd remove this section.